explainaboard/loaders/file_loader.py

```diff
@@ -0,0 +1,221 @@
+        src_names = [field.src_name for field in self._fields]
+        target_names = [field.target_name for field in self._fields]
+        if len(src_names) != len(set(src_names)):
+            raise ValueError("src_name must be unique")
+        if len(target_names) != len(set(target_names)):
+            raise ValueError("target_name must be unique")
+
+    @staticmethod
+    def parse_data(data: Any, field: FileLoaderField) -> Any:
+        if field.parser:
+            return field.parser(data)
+        if field.strip_before_parsing:
+            data = data.strip()
+        dtype = field.dtype
+        if dtype == FileLoaderDType.int:
+            return int(data)
+        if dtype == FileLoaderDType.float:
+            return float(data)
+        if dtype == FileLoaderDType.str:
+            return str(data)
+        if dtype == FileLoaderDType.other:
+            return data
+
+    def generate_id(self, parsed_data_point: dict, sample_idx: int):
+        """generates an id attribute for each data point in place"""
+        if self._use_idx_as_id:
+            parsed_data_point["id"] = str(sample_idx)
+        elif self._id_field_name:
+            if self._id_field_name not in parsed_data_point:
+                raise ValueError(
+                    f"The {sample_idx} data point in system outputs file does not have field {self._id_field_name}"
+                )
+            parsed_data_point["id"] = str(parsed_data_point[self._id_field_name])
+
+    @classmethod
+    def load_raw(self, data: str, source: Source) -> Iterable:
+        """Load data from source and return an iterable of data points. It does not use
+        fields information to parse the data points.
+
+        Args:
+            data (str): base64 encoded system output content or a path for the system output file
+                source: source of data
+        """
+        raise NotImplementedError(
+            "load_raw() is not implemented for the base FileLoader"
+        )
+
+    def load(self, data: str, source: Source) -> Iterable[dict]:
+        """Load data from source, parse data points with fields information and return an
+        iterable of data points.
+        """
+        raw_data = self.load_raw(data, source)
+        parsed_data_points: List[dict] = []
+        for idx, data_point in enumerate(raw_data):
+            parsed_data_point = {}
+
+            for field in self._fields:  # parse data point according to fields
+                parsed_data_point[field.target_name] = self.parse_data(
+                    data_point[field.src_name], field
+                )
+
+            self.generate_id(parsed_data_point, idx)
+            parsed_data_points.append(parsed_data_point)
+        return parsed_data_points
+
+
+@dataclass
+class FileLoaderField:
+    src_name: Union[int, str]  # int for tsv column indices and string for dict keys
+    target_name: str
+    dtype: FileLoaderDType
+    strip_before_parsing: bool = False
```

Comment:
It wasn't immediately obvious to me what this meant. A comment would be helpful.

Also, should we make True be default? It seems this is used far more often than False.

Reply:
I added a docstring for `FileLoaderField`. I made `strip_before_parsing=True` to be the default for `dtype==str` only. I didn't want to make this argument to be True by default because it only works if the src data is a string. If a user ignores this field and left it to be True, they may get an error message about `strip()` which may not be obvious what that means. But I think this should be relatively safe if the target data type is str.